### DISCUSSION TITLE
UdpSocket, can't see packets

### DISCUSSION THREAD

**Discussion by fabracht** [fzyzcjy/flutter_rust_bridge#617] [Q&A]
Hello, I'm trying to send a message with Rust with a UdpSocket, but I don't see any traffic on tcpdump on my host machine.
I created the flutter Counter template project and for now I'm testing out some Rust functionalities that will be part of the full application. My host machine is ubuntu 22.04, I'm running an Android emulator with Android 12 Pixel 4XL API 31 (arm 64bits).
Here's what the api.rs file looks like:

```rust
  use std::net::UdpSocket;
  
  pub fn send_udp_message() -> usize {
      let udpsocket = UdpSocket::bind("10.0.2.15:0").expect("Failed binding to socket");
      
      let sent_bytes = udpsocket
          .send_to(&[0; 10], "10.0.2.2:44581")
          .expect("couldn't send data");
      sent_bytes
  }
  
  pub fn add_one(number: i32) -> i32 {
      println!("Adding one");
      number + 1
  }
```

I call it inside the main.dart `_incrementCounter()` method like this:
```dart
Future<void> _incrementCounter() async {
    var counterPlus = await api.addOne(number: _counter);
    var sent_bytes = await api.sendUdpMessage();

    print("Counter plus is ");
    print(counterPlus);
    print("Sent bytes ");
    print(sent_bytes);

    setState(() {
      _counter = counterPlus;
    });
  }
```

So, I'm just testing out sending a message to a "local" ip address.
According to this documentation [here](https://developer.android.com/studio/run/emulator-networking), 10.0.2.15 is the emulated device network/ethernet interface.
10.0.2.2 is a special alias to the host loopback interface (i.e., 127.0.0.1 on your development machine).

So, I have a good indication that the code is ok because I see the printouts and the value of the counter changes. But when I try to listen to whatever I'm sending on the socket, say by:
```bash
sudo tcpdump -nvXX udp and port 44581
```
on the host machine,
or by trying to listen to it with `nc`, I can't see any packets whatsoever. 
Am I missing something, a configuration somewhere perhaps?

Any help would be appreciated.

Thank you

**Comment by fzyzcjy** [comment]
Hi, this does not seem to be a bug related to this package, but instead a problem with Rust UDP etc.

**Reply by fabracht** [reply]
Ok. I'll do some more research in other places.
